Storefinders: Implement pre and post processing (#9419)
alltheplaces/alltheplaces: docs/API_SPIDER.md

## The store locator has an API

### Common storefinders

Often, a storefinder is a plugin or common software component. To encourage code

> re-use, there are a large number of [pre-built store finders](./locations/storefinders/)

Author: ```suggestion
re-use, there are a large number of [pre-built store finders](../locations/storefinders/)
```

Over and above the [core scrapy spider API](https://docs.scrapy.org/en/latest/topics/spiders.html),
typically these store finders follow a pattern of:

- Specifying an API key or `start_url`
- `pre_process_data` - a method for cleaning or transforming a structure from the API (dict, xpath/dom node) prior to processing.

> - `parse_item` or `post_process_item` - a method for further decorating an item after the primary processing is done. IE, removing invalid names or phone numbers.

Author: ```suggestion
- `parse_item` or `post_process_item` (StructuredDataSpiders) - a method for further decorating an item after the primary processing is done. IE, removing invalid names or phone numbers.
```

While exploring the storefinder you wish to spider, look for indicators or common elements in the existing codebase.
IE: is there an ajax call with a `get_stores` attribute? If so, are there any existing spiders that make similar calls?
This will often lead to very simple, minimal spiders.

Examples:

> - [univeral_store_au](./locations/spiders/universal_store_au.py) - Based on a [closeby](./locations/storefinders/closeby.py) storefinder.

Author: ```suggestion
- [univeral_store_au](../locations/spiders/universal_store_au.py) - Based on a [closeby](../locations/storefinders/closeby.py) storefinder.
```

> - [see_candies](./locations/spiders/see_candies.py) - Based on a [rioseo](./locations/storefinders/closeby.py) storefinder, overriding the default behaviours

Author: ```suggestion
- [see_candies](./locations/spiders/see_candies.py) - Based on a [rioseo](./locations/storefinders/rio_seo.py) storefinder, overriding the default behaviours
```

> - [see_candies](./locations/spiders/see_candies.py) - Based on a [rioseo](./locations/storefinders/rio_seo.py) storefinder, overriding the default behaviours

Author: ```suggestion
- [see_candies](../locations/spiders/see_candies.py) - Based on a [rioseo](../locations/storefinders/rio_seo.py) storefinder, overriding the default behaviours
```
